examples/step-a/step-a.cc

```diff
@@ -0,0 +1,1654 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2000 - 2016 by the deal.II authors
+ *
+ * This file is part of the deal.II library.
+ *
+ * The deal.II library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the deal.II distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Wolfgang Bangerth, University of Heidelberg, 2000
+ *         Daniel Garcia-Sanchez, CNRS, 2018
+ */
+
+// @sect3{Include files}
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/lac/constraint_matrix.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/sparse_direct.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <hdf5.h>
+
+#include <fstream>
+#include <iostream>
+
+// @sect3{Hdf5 library}
+
+namespace hdf5
+{
+  class H5Group
+  {
+  public:
+    enum class FileMode
+    {
+      create,
+      open
+    };
+    enum class H5Boolean
+    {
+      FALSE,
+      TRUE
+    };
+    H5Group(const std::string filename_,
+            const FileMode    filemode_ = FileMode::create);
+    H5Group(const H5Group &parentGroup_, const std::string groupName_);
+    // Rule of five C++
+    H5Group(const H5Group &other);
+    H5Group(H5Group &&other);
+    H5Group &operator=(const H5Group &other);
+    H5Group &operator=(H5Group &&other);
+    ~H5Group();
+    void
+                store_matrix(const std::vector<std::vector<std::complex<double>>> &matrix,
```

Comment:
funny alignment here with the return type separated from the function name :-)

Reply:
I think that this is related to clang-format. The clang-format from the library is different than the format from the examples. I'll put this code in the library, therefore this should be fine. I'll make sure that the format is ok!